### PR TITLE
platforms: allow setting extra flags to be passed to build

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
Untested, not sure how to test (I guess its Python so you never know until it is too late :) )

### CODE REVIEW

File: seL4-platforms/platforms.py
Comment (axel-h) [inline seL4-platforms/platforms.py:113]: This will print
```
extra_settings: {'foo': 'foo1', 'bar': 'bar1'}
```
or
```
extra_settings: {}
```
but I assume we want YAML format:
```
extra_settings:
  foo: foo1
   bar: bar1
```

File: seL4-platforms/platforms.yml
Comment (axel-h) [inline seL4-platforms/platforms.yml:121]: Maybe we should pick a more specific name like `build_params` here. It might also be worth having a note here why these setting are needed and the defaults are not used. Would it make sense to change the defaults instead. This platform is quite new, so there might no be too much dependencies on the default.

File: seL4-platforms/platforms.yml
Comment (axel-h) [inline seL4-platforms/platforms.yml:122]: We have "settings" already in some other YAML files, so we could also use this name here also?

Comment (lsf37) [reply seL4-platforms/platforms.yml:122]: Yes, good idea, that should actually work.

Comment (Ivan-Velickovic) [reply seL4-platforms/platforms.yml]: Is it good now?

File: seL4-platforms/platforms.py
Comment (axel-h) [reply seL4-platforms/platforms.py]: This is still blocking the merge, unless we don't need YAML output here?

Comment (axel-h) [reply seL4-platforms/platforms.py]: Seems we don't need YAML.

**Review by axel-h — APPROVED** [review]
Looks good to me.

**Review by lsf37 — APPROVED** [review]
Cool. I like it with reusing `settings`.

**Comment by Ivan-Velickovic** [comment]
I gave this a test to make sure sel4test ran after merging and to my surprise, adding `Sel4testAllowSettingsOverride` actually does more than I expected.

For example I ran sel4test with `../init-build.sh -DPLATFORM=zynqmp -DSel4testAllowSettingsOverride=1` and `./init-build.sh -DPLATFORM=zynqmp` which I thought would produced the exact same image since no settings are actually being overridden, but it looks like for some reason that the addition of `Sel4testAllowSettingsOverride` now causes sel4test to default to release mode. Not sure if this will be a problem with CI if debug mode is explicitly being turned on.

**Comment by lsf37** [comment]
Hm, it looks like that might influence a whole bunch of settings:
https://github.com/seL4/sel4test/blob/0a488b92bc7a08acc14d49524d1f22656f286627/settings.cmake#L43-L134

That setting is pretty misnamed, it's not an override, but prevents any default settings from applying. This is probably not really what we want to use. We might need to add an actual override that allows you to set things in addition to the defaults.

**Pull request merged**
